### PR TITLE
kubevirt role: deploy presets from demo-content.yaml

### PR DESCRIPTION
**Pull request by lukas-bednar**
Signed-off-by: Lukas Bednar <lbednar@redhat.com>

### CODE REVIEW

**Comment by lukas-bednar** [comment]
Block can not be looped

**Comment by lukas-bednar** [comment]
failing on kubevirt/kubevirt#1103

File: roles/kubevirt/tasks/provision.yml [hunk 1]
```diff
@@ -25,44 +25,13 @@
...
-    dest: /tmp/kubevirt.yaml
-  when: byo_template.stat.exists == True
-
-- name: Create KubeVirt Resources
-  command: kubectl apply -f /tmp/kubevirt.yaml
+- name: Prepare and apply KubeVirt manifests
+  include_tasks: render-and-apply-manifest.yml
+  with_items:
+    - kubevirt.yaml
+    - demo-content.yaml
```
Comment (gbenhaim) [inline roles/kubevirt/tasks/provision.yml]: From where do we get this file?

Comment (lukas-bednar) [reply roles/kubevirt/tasks/provision.yml]: From KubeVirt release page, or from offline templates. depends on env (online / disconnected)

**Comment by rthallisey** [comment]
ci test this

**Comment by lukas-bednar** [comment]
ci test please

**Comment by gbenhaim** [comment]
ci test please

**Comment by gbenhaim** [comment]
There was an error when trying to pull `registry.fedoraproject.org/latest/etcd`.
I've checked with skopeo and the image is available now.

**Comment by lukas-bednar** [comment]
```
error pulling image configuration: Get https://registry.fedoraproject.org/v2/latest/etcd/blobs/sha256:4024b36d7ceaa56c2faf215e7e8ed92181411b5768244022d62a78a344dfbceb: net/http: TLS handshake timeout
``` ?
I will try to re-trigger it again

**Comment by lukas-bednar** [comment]
ci test please

**Comment by gbenhaim** [comment]
```
TASK [kubevirt : Create KubeVirt Resources] ************************************

Thursday 14 June 2018  10:16:20 +0000 (0:00:00.430)       0:20:39.340 ********* 

fatal: [localhost]: FAILED! => {

    "changed": true, 

    "cmd": [

        "kubectl", 

        "apply", 

        "-f", 

        "/tmp/demo-content.yaml"

    ], 

    "delta": "0:00:00.539284", 

    "end": "2018-06-14 10:16:20.759812", 

    "rc": 1, 

    "start": "2018-06-14 10:16:20.220528"

}



STDERR:



The VirtualMachinePreset "windows-server-2012r2" is invalid: []: Invalid value: map[string]interface {}{"metadata":map[string]interface {}{"selector":map[string]interface {}{"matchLabels":map[string]interface {}{"kubevirt.io/os":"win2k12r2"}}, "creationTimestamp":"2018-06-14T10:16:20Z", "uid":"fbb700ff-6fbb-11e8-9ff3-5452c0a8c802", "selfLink":"", "clusterName":"", "annotations":map[string]interface {}{"kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"kubevirt.io/v1alpha1\",\"kind\":\"VirtualMachinePreset\",\"metadata\":{\"annotations\":{},\"name\":\"windows-server-2012r2\",\"namespace\":\"default\",\"selector\":{\"matchLabels\":{\"kubevirt.io/os\":\"win2k12r2\"}}},\"spec\":{\"domain\":{\"clock\":{\"timer\":{\"hpet\":{\"present\":false},\"hyperv\":{},\"pit\":{\"tickPolicy\":\"delay\"},\"rtc\":{\"tickPolicy\":\"catchup\"}},\"utc\":{}},\"cpu\":{\"cores\":2},\"features\":{\"acpi\":{},\"apic\":{},\"hyperv\":{\"relaxed\":{},\"spinlocks\":{\"spinlocks\":8191},\"vapic\":{}}},\"resources\":{\"requests\":{\"memory\":\"2G\"}}}}}\n"}, "name":"windows-server-2012r2", "namespace":"default"}, "spec":map[string]interface {}{"domain":map[string]interface {}{"clock":map[string]interface {}{"timer":map[string]interface {}{"hpet":map[string]interface {}{"present":false}, "hyperv":map[string]interface {}{}, "pit":map[string]interface {}{"tickPolicy":"delay"}, "rtc":map[string]interface {}{"tickPolicy":"catchup"}}, "utc":map[string]interface {}{}}, "cpu":map[string]interface {}{"cores":2}, "features":map[string]interface {}{"acpi":map[string]interface {}{}, "apic":map[string]interface {}{}, "hyperv":map[string]interface {}{"relaxed":map[string]interface {}{}, "spinlocks":map[string]interface {}{"spinlocks":8191}, "vapic":map[string]interface {}{}}}, "resources":map[string]interface {}{"requests":map[string]interface {}{"memory":"2G"}}}}, "apiVersion":"kubevirt.io/v1alpha1", "kind":"VirtualMachinePreset"}: validation failure list:

spec.selector in body is required

spec.domain.devices in body is required
```

**Comment by lukas-bednar** [comment]
@gbenhaim  We are back to original issue kubevirt/kubevirt#1103 :-/

**Comment by tchughesiv** [comment]
@lukas-bednar @rthallisey https://github.com/kubevirt/kubevirt/issues/1103 has been resolved fyi

**Comment by lukas-bednar** [comment]
@tchughesiv I know but we are consuming release, so I am waiting for v0.7.0-alpha.3, it should be out any minute then I will try it out.

**Comment by rthallisey** [comment]
ci test please

**Comment by lukas-bednar** [comment]
@rthallisey it will not work ... kubevirt-ansible is pinned to 0.6.0, and the fix is in 0.7.0-alpha.3

**Comment by rthallisey** [comment]
@lukas-bednar We'll need to bump up CI otherwise this will never pass.

**Comment by lukas-bednar** [comment]
@rthallisey I know, I am just waiting for that release to be published with all [artifacts](https://github.com/kubevirt/kubevirt/releases/tag/v0.7.0-alpha.3) ...

File: roles/kubevirt/tasks/render-and-apply-manifest.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+---
+- name: "Check for {{ manifest_name }}.j2 template in {{ kubevirt_template_dir }}"
+  stat:
+    path: "{{ kubevirt_template_dir }}/{{ manifest_name }}.j2"
+  register: byo_template
```
[the remark this answers is not in the document]
Comment (lukas-bednar) [reply roles/kubevirt/tasks/render-and-apply-manifest.yml]: Bring Your Own

**Comment by lukas-bednar** [comment]
Rebased

**Comment by rthallisey** [comment]
lg2m

**Pull request merged**